robotoff/workers/tasks/import_image.py

```diff
@@ -215,6 +227,68 @@ def run_nutrition_table_object_detection(product_id: ProductIdentifier, image_ur
 }
 
 
+def run_upc_detection(product_id: ProductIdentifier, image_url: str):
+    """Detect the presence of a UPC in an image and determine if it dominates the image (i.e. if it is the primary content in the image)
+    :param product_id: identifier of the product
+    :param image_url: URL of the image to use
+    """
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
+
+    if image is None:
+        logger.info("Error while downloading image %s", image_url)
+        return
+
+    open_cv_image = numpy.array(image)
+
+    source_image = get_source_from_url(image_url)
+
+    pct_area, prediction_class, box = find_image_is_upc(
+        open_cv_image
+    )  # run upc detection
+
+    with db:
+        imagemodel = ImageModel.get_or_none(
+            source_image=source_image, server_type=product_id.server_type.name
+        )
+
+        if not imagemodel:
+            logger.info("Missing image in DB for image %s", source_image)
+            return
+
+        ImagePrediction.create(
```

Comment:
You should check if an `ImagePrediction` already exists, and if it does use this image_prediction instead. We want our detection jobs to be idempotent.
See how `run_object_detection_model` works.

Reply:
Would this look like this/what should I do with the prediction if it already exists? : 
```
if (
        existing_image_prediction := ImagePrediction.get_or_none(
            image=image_model, model_name="upc_opencv"
        )
        ) is not None:
            pass
        else:
            ImagePrediction.create(
                image=image_model,
                type="upc_image",
                model_name="upc_opencv",
                model_version="upc_opencv_1.0",
                data={
                    "bounding_box": box.tolist(),
                    "area": area,
                    "class": prediction_class.value,
                },
                max_confidence=None,
            )
```
Trying to model similarly after run_object_detection_model but not exactly sure what to do here.